php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1532 @@
+    return b.<ForStatementHeader>nonterminal()
+      .is(f.forStatementHeader(
+        b.token(PHPKeyword.FOR), b.token(PHPPunctuator.LPARENTHESIS),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.SEMICOLON),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.SEMICOLON),
+        b.optional(FOR_EXPR()),
+        b.token(PHPPunctuator.RPARENTHESIS)
+      ));
+  }
+
+  public SeparatedList<ExpressionTree> FOR_EXPR() {
+    return b.<SeparatedList<ExpressionTree>>nonterminal(PHPLexicalGrammar.FOR_EXPR)
+      .is(f.forExpr(
+        EXPRESSION(),
+        b.zeroOrMore(f.newTuple12(b.token(PHPPunctuator.COMMA), EXPRESSION()))
+      ));
+  }
+
+  public ForEachStatementTree FOREACH_STATEMENT() {
+    return b.<ForEachStatementTree>nonterminal(PHPLexicalGrammar.FOREACH_STATEMENT)
+      .is(b.firstOf(
+          f.forEachStatement(FOREACH_STATEMENT_HEADER(), STATEMENT()),
+          f.forEachStatementAlternative(
+            FOREACH_STATEMENT_HEADER(),
+            b.token(PHPPunctuator.COLON),
+            b.zeroOrMore(INNER_STATEMENT()),
+            b.token(PHPKeyword.ENDFOREACH),
+            EOS()
+          ))
+      );
+  }
+
+  public ForEachStatementHeader FOREACH_STATEMENT_HEADER() {
+    return b.<ForEachStatementHeader>nonterminal()
+      .is(f.forEachStatementHeader(
+        b.token(PHPKeyword.FOREACH), b.token(PHPPunctuator.LPARENTHESIS),
+        EXPRESSION(), b.token(PHPKeyword.AS),
+        b.optional(f.newTuple10(FOREACH_VARIABLE(), b.token(PHPPunctuator.DOUBLEARROW))), FOREACH_VARIABLE(),
+        b.token(PHPPunctuator.RPARENTHESIS)
+      ));
+  }
+
+  public ExpressionTree FOREACH_VARIABLE() {
+    return b.<ExpressionTree>nonterminal().is(
+      b.firstOf(REFERENCE_VARIABLE(), MEMBER_EXPRESSION(), LIST_EXPRESSION())
+    );
+  }
+
+  public ThrowStatementTree THROW_STATEMENT() {
+    return b.<ThrowStatementTree>nonterminal(PHPLexicalGrammar.THROW_STATEMENT)
+      .is(f.throwStatement(b.token(PHPKeyword.THROW), EXPRESSION(), EOS()));
+  }
+
+  public EmptyStatementTree EMPTY_STATEMENT() {
+    return b.<EmptyStatementTree>nonterminal(PHPLexicalGrammar.EMPTY_STATEMENT)
+      .is(f.emptyStatement(b.token(PHPPunctuator.SEMICOLON)));
+  }
+
+  public ReturnStatementTree RETURN_STATEMENT() {
+    return b.<ReturnStatementTree>nonterminal(PHPLexicalGrammar.RETURN_STATEMENT)
+      .is(f.returnStatement(b.token(PHPKeyword.RETURN), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public ContinueStatementTree CONTINUE_STATEMENT() {
+    return b.<ContinueStatementTree>nonterminal(PHPLexicalGrammar.CONTINUE_STATEMENT)
+      .is(f.continueStatement(b.token(PHPKeyword.CONTINUE), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public BreakStatementTree BREAK_STATEMENT() {
+    return b.<BreakStatementTree>nonterminal(PHPLexicalGrammar.BREAK_STATEMENT)
+      .is(f.breakStatement(b.token(PHPKeyword.BREAK), b.optional(EXPRESSION()), EOS()));
+  }
+
+  public TryStatementTree TRY_STATEMENT() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)